index.js

```diff
@@ -339,3 +339,65 @@ module.exports.random = {
     return s * Math.pow(2, (-64 - e))
   }
 }
+
+/**
+ * Utils for simple signing and verification of HTTPS requests
+ **/
+module.exports.https = {
+  /**
+   * Sign the `message` with the given `secretKey`.
+   *
+   * @method
+   * @param {any} headers - headers containing the properties to sign
+   * @param {string} keyId - an opaque string that the server can use to look up the component they need to validate the signature
+   * @param {string} secretKey - secret key to sign the message
+   * @returns {string}
+   */
+  ed25519Sign: function (headers /* any */, keyId /* string */, secretKey /* string */) {
+    if (!secretKey) throw new Error('secret key is required')
+    if (!keyId) throw new Error('key id is required')
+    if (JSON.stringify(headers) === '{}') throw new Error('headers are required')
```

Comment:
should this be `if (!headers || JSON.stringify(headers) === '{}')...` ?

Reply:
also i slightly prefer `Object.keys(headers).length === 0` to check for an empty object